data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java

```diff
@@ -0,0 +1,262 @@
+            .map(MessageReader::toEvent)
+            .onFailure(context::failNow)
+            .onSuccess(event -> {
+
+              if (request.path().equals(PATH_SERVICE_1)) { // service 1 receives requestEvent
+                context.verify(() -> {
+                  assertThat(event).isEqualTo(requestEvent);
+                  checkpoints.flag(); // 2
+                });
+
+                VertxMessageFactory.createWriter(request.response()).writeBinary(responseEvent);
+              }
+              if (request.path().equals(PATH_SERVICE_2)) { // service 2 receives responseEvent
+                context.verify(() -> {
+                  assertThat(event).isEqualTo(responseEvent);
+                  checkpoints.flag(); // 3
+                });
+              }
+            }))
+        .listen(SERVICE_PORT, "localhost");
+
+    // do the request to the Broker receiver
+    final var request = vertx.createHttpClient()
+        .post(INGRESS_PORT, "localhost", format("%s/%s", BROKER_NAMESPACE, BROKER_NAME))
+        .exceptionHandler(context::failNow)
+        .handler(response -> context.verify(() -> {
+          assertThat(response.statusCode()).isLessThan(300); // verify it's a 2xx response
+          checkpoints.flag(); // 1
+        }));
+
+    VertxMessageFactory.createWriter(request).writeBinary(requestEvent);
+  }
+
+  @AfterAll
+  public static void teardown(final VertxTestContext context) {
+
+    // TODO figure out why shutdown times out Vertx context even with timeout increased.
+    // kafkaCluster.shutdown();
+
+    if (!dataDir.delete()) {
+      dataDir.deleteOnExit();
+    }
+
+    context.completeNow();
+  }
+
+  private static void setUpKafkaCluster() throws IOException {
+
+    dataDir = File.createTempFile("kafka", "kafka");
+
+    kafkaCluster = new KafkaCluster()
+        .withPorts(ZK_PORT, KAFKA_PORT)
+        .deleteDataPriorToStartup(true)
+        .addBrokers(NUM_BROKERS)
+        .usingDirectory(dataDir)
+        .startup();
+
+    kafkaCluster.createTopic(TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+  }
+
+  private static void setUpDispatcher(final Vertx vertx) {
+
+    final ConsumerRecordOffsetStrategyFactory<String, CloudEvent>
+        consumerRecordOffsetStrategyFactory = ConsumerRecordOffsetStrategyFactory.create();
+
+    final var consumerConfigs = new Properties();
+    consumerConfigs.put(BOOTSTRAP_SERVERS_CONFIG, format("localhost:%d", KAFKA_PORT));
+    consumerConfigs.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    consumerConfigs.put(VALUE_DESERIALIZER_CLASS_CONFIG, CloudEventDeserializer.class);
+
+    final var producerConfigs = producerConfigs();
+
+    final var consumerVerticleFactory = new HttpConsumerVerticleFactory(
+        consumerRecordOffsetStrategyFactory,
+        consumerConfigs,
+        vertx.createHttpClient(),
+        vertx,
+        producerConfigs
+    );
+
+    brokersManager = new BrokersManager<>(
```

Comment:
Could you return this value more than setting it here? So in future we could reuse this method and promote it to static